CenterForOpenScience/ember-osf-web
# [My Registrations][ENG-2542] [ENG-2584] [ENG-1686] Show project's draft reg tab for all and update draft reg card

- Ticket: [ENG-2542] [ENG-2584] [ENG-1686]
- Feature flag: n/a

## Purpose
- Show the Draft registrations tab for all contributors on the project
- Update the Draft Registration Card component to show/hide the `Edit`, `Remove`, and `Review` buttons depending on the user's permissions on that draft
- Update the Draft Registration Card to show the provider the draft is going to be submitted to

## Summary of Changes
### Project Draft Registrations tab
- Check if the current user has at least read-permissions to view the drafts tab
- Only show the option to add a create a new draft to admin-contributors
### Draft Registration Card
- Add a new line indicating the provider that the draft is being submitted to
- Only show `Remove` button to admin contributors
- Show `Edit` button to admins and read-write contributors
- Show `Review` button to all contributors
- Rearrange buttons so they are in a more predictable location
- Use DeleteButton component for removing draft and update how icons are handled in this component
- Update tests

## Screenshots
### Project Draft Registrations tab
- Read-only or Read-write; No drafts
![image](https://user-images.githubusercontent.com/51409893/112529153-57e43700-8d7b-11eb-9eb1-331d21595f16.png)
- Admin; No drafts
![Screen Shot 2021-03-25 at 3 06 18 PM](https://user-images.githubusercontent.com/51409893/112529448-a8f42b00-8d7b-11eb-8a26-bd7e2da04857.png)
### Draft Registration Card
![Screen Shot 2021-04-19 at 2 30 31 PM](https://user-images.githubusercontent.com/51409893/115285384-ccf92100-a11b-11eb-99e3-6a50cdad4895.png)



## Side Effects
- I don't think there will be any side-effects with this PR, but it is possible that a user that is an admin on the project is only a read-write, or read-only contributor for a draft (projects and drafts can have different contributors with different permissions)

## QA Notes
- Please verify that the language and permissions all make sense. As mentioned in the `Side Effects`, permissions between project and draft registrations can now deviate, so please keep that in mind when testing.


[ENG-2542]: https://openscience.atlassian.net/browse/ENG-2542
[ENG-2584]: https://openscience.atlassian.net/browse/ENG-2584
[ENG-1686]: https://openscience.atlassian.net/browse/ENG-1686
